### PR TITLE
Tisserand plotting utilities

### PR DESCRIPTION
**Pull request by jorgepiloto**
Solves for #925 as suggested from feedback. Opening this as draft since the following tasks still require to be solved:

- [x] Solve for small plotting bugs regarding upper bounding limits
- [x] Add proper image tests
- [x] Document this module and add notebook examples

### CODE REVIEW

File: src/poliastro/plotting/tisserand.py [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+    # Generate Tisserand
+    tp = TisserandPlotter()
+
+    for ss in SS_BODIES:
+
+        # This example works fine
+        tp.plot(ss, (1, 10) * u.km / u.s, num_contours=10)
+
+        # TODO: BUG APPEARS IF THE UPPER BOUND FOR VINF > 10 KM/S
+        #tp.plot(ss, (1, 11) * u.km / u.s, num_contours=11)
```
Comment (astrojuanlu) [inline src/poliastro/plotting/tisserand.py:139]: I can reproduce the issue. It looks like there's an x value going to 0, or similar.

**Comment by astrojuanlu** [comment]
I think the problem is this:

![Screenshot from 2020-04-27 00-52-35](https://user-images.githubusercontent.com/316517/80321912-6d767700-8821-11ea-893a-9d77b82c84bd.png)

For a given value of α, the semimajor axis starts positive (elliptic orbit), then tends to infinity when `1 - v_∞^2 - 2 v_∞ cos(α)` approaches zero, then comes from minus infinity (hyperbolic orbit). I think it does not make sense to go past the "limit" v_∞, which for α = 0 is, in non-dimensional variables, is:

```
In [8]: solve(Eq(x ** 2 + 2 * x - 1, 0))                                                                                                                                                                                                      
Out[8]: [-1 + sqrt(2), -sqrt(2) - 1]

In [9]: N(_[0])                                                                                                                                                                                                                               
Out[9]: 0.414213562373095
```

Or around ~11 km / s :wink:

**Comment by jorgepiloto** [comment]
After a little bit of testing, I found that for angle values near zero, the plot
starts to show this behaviour. However, the original paper claims that the
method is valid for any angle within [0, pi]. Hope to continue working on this
and discover what is going on...

**Comment by jorgepiloto** [comment]
I see... But then, how is it possible for the author of the paper to plot the lines for v_inf=11km without running into singularities? I also noticed small differences between the values obtained with this and the ones from the paper but my reasoning is that the values for the position and velocity of the planets are just "slightly different".

**Comment by astrojuanlu** [comment]
> But then, how is it possible for the author of the paper to plot the lines for v_inf=11km without running into singularities? 

Which paper exactly? I'm looking at figure 3.4 from David Palma thesis and it goes up to 9 km/s.

**Comment by jorgepiloto** [comment]
Ops! Sorry for this, I should have posted a pic of the figure I am looking at. All credits go to David Falcato Fialho Palma:

tisserand

Previous figure is located in the appendix A together with other tisserand images.

**Comment by astrojuanlu** [comment]
Thinking about this again. The text confirms that hyperbolic orbits are out of scope:

> Tisserand graphs come in many shapes, depending on their purpose, but the one that is most useful to us is the apoapsis vs periapsis (R a vs R p ) formulation (see figure 3.4). It allows for a clearer vision of the spacecraft orbit, despite not being suitable for representing hyperbolic trajectories but these won’t be studied in this work.

I wonder if you are right:

> my reasoning is that the values for the position and velocity of the planets are just "slightly different".

**Comment by jorgepiloto** [comment]
As discussed privately, there seems to be some negative value coming from the computation of the `a_sc` in which the relationship between `V_inf` and `cos(alpha)` returns a negative value. By forcing the denominator to be always positive with `np.abs()` the issue seems to be solved.

However, previous "hack" requires further investigation. In addition, although some figures look like the ones from different papers, values are close but not the same.

**Comment by jorgepiloto** [comment]
I have been working on this during the last days' but got stuck while thinking about the following: imagine as user I want to plot all the three main Tisserand configurations (`E_vs_rp`, `T_vs_rp` and `rp_vs_ra`). I thought about the possibility of having three internal axes, one for each kind of plot, but although this works fine, all of them are displayed one created...

Algorithm seems to be working fine and outputs same lines for different examples found in academical papers:

<p float="left">
  
   
</p>

**Comment by astrojuanlu** [comment]
Sorry for the long delay reviewing this! I will leave some comments. In the meanwhile, this will also need a rebase.

File: src/poliastro/plotting/tisserand.py [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+        # Tisserand kind collection
+        self.type_collection = ["energy", "period", "apsis"]
+
+        # Prellocate internal data lines
+        self.energy_lines = []
+        self.period_lines = []
+        self.apsis_lines = []
+
+    @property
+    def rp_vs_ra(self):
```
Comment (astrojuanlu) [inline src/poliastro/plotting/tisserand.py:40]: I suggest removing all the logic that mutates the axes, and instead offering something like this:

```
tisserand = TisserandPlotter()
tisserand = TisserandPlotter(kind=TisserandKind.ENERGY)  # Same as above, default kind
tisserand = TisserandPlotter(kind=TisserandKind.PERIOD)
tisserand = TisserandPlotter(kind=TisserandKind.APSIS)
```

How does that sound @jorgepiloto?

Comment (jorgepiloto) [reply src/poliastro/plotting/tisserand.py:40]: I tried this implementation and really like it! Let me push new changes before cleaning git commit history :+1:

**Comment by jorgepiloto** [comment]
Sorry for this late contribution. I just added some new changes:

* A new `plot_line` method for better showing the final desired gravity assist path. This would be the "`plot_trajectory`" but in Tisserand figures.
* Documented the module by adding it to the API reference and including a jupytext notebook example.

**Comment by astrojuanlu** [comment]
Needs `tox -e reformat`

**Comment by astrojuanlu** [comment]
Both tests failures come from our dear `rv2coe` docstring. Merging!

**Pull request merged**